src/init.cpp

```diff
@@ -170,12 +171,24 @@ static fs::path GetPidFile(const ArgsManager& args)
 #else
         tfm::format(file, "%d\n", getpid());
 #endif
+        g_generated_pid = true;
         return true;
     } else {
         return InitError(strprintf(_("Unable to create the PID file '%s': %s"), fs::PathToString(GetPidFile(args)), SysErrorString(errno)));
     }
 }
 
+static void RemovePidFile(const ArgsManager& args)
+{
+    if (!g_generated_pid) return;
+    auto pid_path = GetPidFile(args);
+    if (std::error_code error; !fs::remove(pid_path, error)) {
+        std::string msg{error ? error.message() : "File does not exist"};
+        LogPrintf("Unable to remove PID file %s: %s\n", fs::quoted(fs::PathToString(pid_path)), msg);
```

Comment:
nit: I'm not sure how important the quoting is, could just use parentheses to make it a bit more concise?

```suggestion
        LogPrintf("Unable to remove PID file (%s): %s\n", fs::PathToString(pid_path), msg);
```